packages/mui-varnish/src/colors.ts

```diff
@@ -0,0 +1,163 @@
+import { Dictionary, Indexable } from './util';
+
+export class RGB {
+  constructor(public r: number, public g: number, public b: number) {}
+
+  toString() {
+    return `rgb(${this.r},${this.g},${this.b})`;
+  }
+}
+
+// convert a hex color string to a RGB
+export function hexToRgb(hex: string): RGB {
+  // Expand shorthand form (e.g. "03F") to full form (e.g. "0033FF")
+  const shorthandRegex = /^#?([\da-f])([\da-f])([\da-f])$/i;
+  hex = hex.replace(shorthandRegex, (_, r, g, b) => {
+    return r + r + g + g + b + b;
+  });
+  const result = /^#?([\da-f]{2})([\da-f]{2})([\da-f]{2})$/i.exec(hex);
+  return result
+    ? new RGB(parseInt(result[1], 16), parseInt(result[2], 16), parseInt(result[3], 16))
+    : new RGB(0, 0, 0);
+}
```

Comment:
I like avoiding regular expressions where possible. I also think we're probably doing a lot more work here than we need to. We should do our best to convert it to an RGB value. If we can't we should throw an exception. And if the resulting color isn't what the developer expects, they should notice and fix the input.

Oh, and sorry that my example code isn't valid TypeScrfipt. It's been a bit :).

```
func expandShortHex(hex: string) string {
   return [ hex[0], hex[0], hex[1], hex[1], hex[2], hex[2] ].join('');
}

func hexToRGB(hex: string): RGB {
  const sanitizedHex = hex[0] == '#' ? hex.slice(1) : hex
  const isShorthand = hex.length < 6
  const expandedHex = isShorthand ? expandShorthand(hex) : hex
  if expandedHex.length !== 6 {
     throw Error(`invalid hex value: ${hex}`)
  }
  return RBG(expandedHex.slice(0, 2), expandedHex.slice(2, 4), expandedHex.slice(4, 6))
}
```

There might be an off by one error or two buried in here 😅 .

Reply:
Oh, and if it's easy to add tests. I'd consider adding a unit test or two for this, as the I/O should make some unit tests relatively straightforward to compose.